File: Java/src/main/java/Exceptions/NotRegisteredException.java
package Exceptions;

import java.lang.reflect.Type;

/**
 * Created by user on 22/9/2015.
 */
public class NotRegisteredException extends RuntimeException {
    public NotRegisteredException(Type contract) {
        super("Cannot resolve interface or abstract class if it has not been registered:" + contract.getTypeName());
    }
}

File: Container/Source/Exceptions/NotRegisteredException.cs
using System;

namespace IoC.Source.Exceptions
{
    /// <summary>
    /// Indicates that an interface or an abstract class is trying to be resolved that is not registered with the container.
    /// </summary>
    internal class NotRegisteredException : Exception
    {
        public NotRegisteredException(Type contract)
            : base($"Cannot resolve interface or abstract class if it has not been registered: {contract.FullName}")
        {
        }
    }
}
